gpt_engineer/preprompts/improve

```diff
@@ -1,50 +1,15 @@
 
-Take requests for changes to the supplied code, and then you MUST
-1. (planning) Think step-by-step and explain the needed changes. Don't include *block*s in this part of your response, only describe code changes.
-2. (generate diffs) Write changes to the code in the unified "git diff" syntax; The unambiguous and compact format that git uses to display the exact difference between two versions of one file.
-Finish by the exact sentence: These changes yield a fully functional program according to the instructions.
+FILE_FORMAT
 
-Example of git diff on an existing file:
+As far as compatible with the user request, start with the "entrypoint" file, then go to the ones that are imported by that file, and so on.
+Please note that the code should be fully functional. No placeholders.
 
-```diff
---- example.txt
-+++ example.txt
-@@ -6,3 +6,4 @@
-     line content A
-     line content B
-+    new line added
--    original line X
-+    modified line X with changes
-@@ -26,4 +27,5 @@
-         condition check:
--            action for condition A
-+            if certain condition is met:
-+                alternative action for condition A
-         another condition check:
--            action for condition B
-+            modified action for condition B
-```
+Follow a language and framework appropriate best practice file naming convention.
+Make sure that files contain all imports, types etc.  The code should be fully functional. Make sure that code in different files are compatible with each other.
+Ensure to implement all code, if you are unsure, write a plausible implementation.
+Include module dependency or package manager dependency definition file.
+Before you finish, double check that all parts of the architecture is present in the files.
 
-Example of a git diff creating a new file:
-
-```diff
---- /dev/null
-+++ new_file.txt
-@@ -0,0 +1,3 @@
-+First example line
-+
-+Last example line
-```
--A program will apply the diffs you generate exactly to the code, so diffs must be precise and unambiguous!
--
--Every diff must be fenced with ```CONTENT OF EDIT BLOCK``` with the correct code language.
--
--The file names at the beginning of a diff, (lines starting with --- and +++) is the relative path to the file before and after the diff.
--
--LINES TO BE REMOVED (starting with single -) AND LINES TO BE RETAIN (no starting symbol) HAVE TO REPLICATE THE DIFFED HUNK OF THE CODE EXACTLY LINE BY LINE.
--NEVER OMIT ANY WHITESPACE NEEDED TO KEEP THE SUCH LINES IDENTICAL.
--KEEP THE DIFF HUNKS (A HUNK STARTS WITH "@@ LINE NUMBER INSTRUCTIONS @@" AS COMPACT AS POSSIBLE; IF POSSIBLE KEEP HUNKS SHORTER THAN 19 LINES BY BREAKING THEM UP INTO MULTIPLE HUNKS.
--AVOID STARTING A HUNK WITH AN EMPTY LINE
+When you are done, write finish with "this concludes a fully working implementation".
```

Comment:
Unsure about this ... is never a fully working implementation because its only outputting a change? is it sometimes a fully working implementation and sometimes not a fully working implementation because its doing the work in chunks ?

Reply:
I've adopted a flow that's very similar to the generate syntax, to make the diff syntax less "lazy" and in my tests, this prompting is a lot better. The "this concludes a fully working implementation" part is some kind of hack, which makes the LLM more prone to actually write code, rather than a delimiter. Potentially it could be moderated to be more in the "diff" context, but I think it does the job.